xdc-plugin/xdc.cc

```diff
@@ -154,6 +154,107 @@ struct GetIOBanks : public Pass {
 	std::function<const BankTilesMap&()> get_bank_tiles;
 };
 
+struct GetNets : public Pass {
+	GetNets() : Pass("get_nets", "Print matching nets") {}
+
+	void help() override
+	{
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n");
+		log("   get_nets [-quiet] [-filter filter_expression] <net_selection> \n");
+		log("\n");
+		log("Get matching nets\n");
+		log("\n");
+		log("Print the output to stdout too. This is useful when all Yosys is executed.\n");
+		log("\n");
+		log("    -filter\n");
+		log("        Name and value of attribute to be taken into account.\n");
+		log("        e.g. -filter { attr == \"true\" }\n");
+		log("\n");
+		log("    -quiet\n");
+		log("        Don't print the result of the execution to stdout.\n");
+		log("\n");
+		log("    <net_selection>\n");
+		log("        Selection of net name. Default are all nets in the design.\n");
+		log("\n");
+
+	}
+
+	void execute(std::vector<std::string> args, RTLIL::Design* design) override
+	{
+		RTLIL::Module* top_module = design->top_module();
+		if (top_module == nullptr) {
+			log_cmd_error("No top module detected\n");
+		}
+
+		size_t argidx;
+		std::pair<std::string, std::string> filter;
+		bool is_quiet = false;
+		bool is_filter = false;
+
+		// Parse command arguments
+		for (argidx = 1; argidx < args.size(); argidx++) {
+		    std::string arg = args[argidx];
+		    if (arg == "-quiet") {
+			is_quiet = true;
+			continue;
+		    }
+		    if (arg == "-filter" and argidx + 1 < args.size()) {
+			std::string filter_arg = args[++argidx];
+			filter_arg.erase(std::remove_if(filter_arg.begin(), filter_arg.end(), isspace), filter_arg.end());
+			auto separator = filter_arg.find("==");
```

Comment:
This is a pretty narrow version of "-filter", should detect and generate errors and/or warnings if the limited filter support is exceeded by the call.

Reply:
Changed the implementation to easily extend for proper filter support. Added a warning in the current narrow version.